ndarray/ndarray-core/src/commonMain/kotlin/io/kinference/ndarray/extensions/onehot/OneHotExtension.kt

```diff
@@ -0,0 +1,34 @@
+package io.kinference.ndarray.extensions.onehot
+
+import io.kinference.ndarray.arrays.*
+
+internal fun NumberNDArrayCore.getOneHotIndices(depth: Int) = when (this) {
+    is IntNDArray -> getOneHotIndices(this, depth)
+    is FloatNDArray -> getOneHotIndices(this, depth)
+    is DoubleNDArray -> getOneHotIndices(this, depth)
+    is LongNDArray -> getOneHotIndices(this, depth)
+    is ShortNDArray -> getOneHotIndices(this, depth)
+    is ByteNDArray -> getOneHotIndices(this, depth)
+    is UIntNDArray -> getOneHotIndices(this, depth)
+    is UShortNDArray -> getOneHotIndices(this, depth)
+    is UByteNDArray -> getOneHotIndices(this, depth)
+    is ULongNDArray -> getOneHotIndices(this, depth)
+    else -> error("OneHot indices array must have numeric data type, current type: $type")
+}
+
+suspend fun oneHot(indices: NumberNDArrayCore, depth: Int, values: NDArrayCore, axis: Int = -1): NDArrayCore {
```

Comment:
I think it should be extension for `NumberCoreNDArray` like in TFJS

Reply:
Also I think we need API with `onValue` and `offValue` like in primitive, it's more convenient for end-user